src/api/translation-api.ts

```diff
@@ -1,28 +1,46 @@
 import { debug, setFailed } from '@actions/core';
+import Axios, { AxiosRequestConfig } from 'axios';
+import { v4 } from 'uuid';
 import { AvailableTranslations } from '../abstractions/available-translations';
 import { TranslationResult, TranslationResults, TranslationResultSet } from '../abstractions/translation-results';
-import { v4 } from 'uuid';
-import Axios, { AxiosRequestConfig } from 'axios';
 import { TranslatorResource } from '../abstractions/translator-resource';
-import { chunk } from '../helpers/utils';
 import { toResultSet } from '../helpers/api-result-set-mapper';
+import { batch, chunk } from '../helpers/utils';
 
 /**
 * https://docs.microsoft.com/azure/cognitive-services/translator/language-support#translate
-*/ 
+*/
 export const getAvailableTranslations = async (): Promise<AvailableTranslations> => {
     const apiUrl = 'https://api.cognitive.microsofttranslator.com/languages';
     const query = 'api-version=3.0&scope=translation';
     const response = await Axios.get<AvailableTranslations>(`${apiUrl}?${query}`);
 
     return response.data;
-}
+};
 
 export const translate = async (
     translatorResource: TranslatorResource,
     toLocales: string[],
-    translatableText: Map<string, string>): Promise<TranslationResultSet | undefined> => {
+    translatableText: Map<string, string>,
+    filePath: string): Promise<TranslationResultSet | undefined> => {
     try {
+        // Current Azure Translator API rate limit
+        // https://docs.microsoft.com/azure/cognitive-services/translator/request-limits#character-and-array-limits-per-request
+        const apiRateLimit = 10000;
+        const numberOfElementsLimit = 100;
+
+        const validationErrors: string[] = [];
+        translatableText.forEach((value, key) => {
+            const valueStringifiedLength = JSON.stringify(value).length;
+            if (valueStringifiedLength > apiRateLimit) {
+                validationErrors.push(`Text for key '${key}' in file '${filePath}' is too long (${valueStringifiedLength}). Must be ${apiRateLimit} at most.`);
+            }
+        });
+        if (validationErrors.length) {
```

Comment:
Is this a new early exit? If so, do you anticipate it ever actually happening? Is there anyway to avoid that, and somehow buffer partial calls? That would probably be a separate PR - just thinking out loud.

Reply:
It can happen if one have more than 10000 chars in a single resource text. I believe it is better to fail fast here without calling translation service.
Yes, we can do partial calls, but it would require to implement even more sophisticated buffering strategy here. In other words, we need to know how to split single resource text. Perhaps by '.' char? I believe that letting users know that they have some too long text, is good enough solution for now.